lib/components/home/artist_rails/artist_rail.js

```diff
@@ -12,17 +12,42 @@ import ArtistCard from './artist_card'
 import Separator from '../../separator'
 
 class ArtistRail extends React.Component {
+  state: {
+    artists: [Object],
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      artists: []
+    }
+  }
 
   componentDidMount() {
     this.props.relay.setVariables({ fetchContent: true })
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.rail.results) {
+      this.setState({ artists: nextProps.rail.results.slice(0) })
```

Comment:
It is actually not really needed in this location, I do it below in `handleFollowChange` again before making a change to the array. I’ll get rid of it here.